mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSModelImportAsNode.kt

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.model.mpsadapters
+
+import org.jetbrains.mps.openapi.model.SModel
+import org.modelix.model.api.BuiltinLanguages
+import org.modelix.model.api.IChildLink
+import org.modelix.model.api.IConcept
+import org.modelix.model.api.IConceptReference
+import org.modelix.model.api.IDeprecatedNodeDefaults
+import org.modelix.model.api.INode
+import org.modelix.model.api.INodeReference
+import org.modelix.model.api.IProperty
+import org.modelix.model.api.IReferenceLink
+import org.modelix.model.area.IArea
+
+class MPSModelImportAsNode(val importedModel: SModel, val importingModel: SModel) : IDeprecatedNodeDefaults {
+    override fun getArea(): IArea =
+        MPSArea(importingModel.repository)
+
+    override val isValid: Boolean
+        get() = TODO("Not yet implemented")
+    override val reference: INodeReference
+        get() = TODO()
+    override val concept: IConcept
+        get() = BuiltinLanguages.MPSRepositoryConcepts.ModelReference
+    override val parent: INode
+        get() = MPSModelAsNode(importingModel)
+
+    override fun getConceptReference(): IConceptReference? {
+        return concept.getReference()
+    }
+
+    override val allChildren: Iterable<INode>
+        get() = emptyList()
+
+    override fun removeChild(child: INode) {
+        throw UnsupportedOperationException("Concept $concept does not have children")
+    }
+
+    override fun getChildren(link: IChildLink): Iterable<INode> {
+        throw UnsupportedOperationException("Concept $concept does not have children")
+    }
+
+    override fun moveChild(role: IChildLink, index: Int, child: INode) {
+        throw UnsupportedOperationException("Concept $concept does not have children")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConcept?): INode {
+        throw UnsupportedOperationException("Concept ${this.concept} does not have children")
+    }
+
+    override fun addNewChild(role: IChildLink, index: Int, concept: IConceptReference?): INode {
+        throw UnsupportedOperationException("Concept ${this.concept} does not have children")
+    }
+
+    override fun getReferenceTarget(link: IReferenceLink): INode {
+        check(link.conformsTo(BuiltinLanguages.MPSRepositoryConcepts.ModelReference.model)) {
+            "Unknown reference link '$link'"
+        }
+        return MPSModelAsNode(importedModel)
+    }
+
+    override fun getReferenceTargetRef(role: IReferenceLink): INodeReference? {
+        return getReferenceTarget(role)?.reference
+    }
+
+    override fun setReferenceTarget(link: IReferenceLink, target: INode?) {
+        if (link.conformsTo(BuiltinLanguages.MPSRepositoryConcepts.ModelReference.model)) {
+            throw UnsupportedOperationException("ModelReference.model is read only.")
+        }
+        throw IllegalArgumentException("Unknown link '$link'")
```

Comment:
We have a wild mixture of effective IllegalStateExceptions, IllegalArgumentExceptions and UnsupportedOperationExceptions here. Is that intentional?